src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3494,8 +3496,17 @@ private ConversionOperatorDeclarationSyntax TryParseConversionOperatorDeclaratio
 
                 if (this.CurrentToken.ContextualKind == SyntaxKind.ExtensionKeyword)
                 {
-                    this.Reset(ref point);
-                    return null;
+                    // implicit extension.operator x
+                    // implicit extension::X.operator x
+                    // implicit extension<X>.operator x
+                    //
+                    // All of these should be treated as a conversion operator, with `extension` as part of an explicit
+                    // interface name.
+                    if (this.PeekToken(1).Kind is not SyntaxKind.ColonColonToken and not SyntaxKind.DotToken and not SyntaxKind.LessThanToken)
```

Comment:
>this.PeekToken(1).Kind is not SyntaxKind.ColonColonToken and not SyntaxKind.DotToken and not SyntaxKind.LessThanToken [](http://example.com/codeflow?start=24&length=117)

It looks like this condition is used in two places and we want them to stay similar as the code evolves. Consider extracting a shared helper with a meaningful name.